File: server-c/server/options.c
#include "./options.h"

// Define server options.
struct Options options = { 0, 0, 0, 0, NULL };

// Define options file instance.
static li_handler* optionsFile;

// Initialize server options.
void optionsInit() {
  // Load options file.
  optionsFile = li_init("server.ini");
  // Load options values.
  options = (struct Options) {
    strtol(li_find_one(optionsFile, "Connection", "ServerMode"), NULL, 10),
    strtol(li_find_one(optionsFile, "Options", "Register"), NULL, 10),
    strtol(li_find_one(optionsFile, "Options", "MultiScores"), NULL, 10),
    strtol(li_find_one(optionsFile, "Options", "NoticeMode"), NULL, 10),
    li_find_one(optionsFile, "Options", "NoticeText"),
  };
}

// Close server options.
void optionsClose() {
  // Close options file.
  li_close(optionsFile);
}

File: server-js/server/options.js
// Define server options.
export let options = {
  // Allow unregistered users to be registered at the login screen.
  register: undefined,
  // Allow users to have mutiple scores (and replays) in the global rankings.
  // Don't change once the database has already been created.
  multiScores: undefined,
  // Define notice message text.
  // 0. No message text.
  // 1. Custom message text.
  noticeMode: undefined,
  noticeText: undefined,
};

// Initialize server options.
export const optionsInit = () => {
  options = {
    register: 1,
    multiScores: 0,
    noticeMode: 1,
    noticeText: 'Jewelry Master - Server Emulator by Hipnosis.',
  }
}